### PR TITLE
Check unicode normalization

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

File: util/check_characters.py [hunk 1]
```diff
@@ -19,11 +19,14 @@
...
 import sys
 
+import unicodedata
+
 def main():
     parser = argparse.ArgumentParser()
 
     parser.add_argument("-csv", "--csv-files", help="Str. Filenames as a comma separated list", required=True)
     parser.add_argument("-alpha", "--alphabet-format", help="Bool. Print in format for alphabet.txt", action="store_true")
+    parser.add_argument("-unicode", "--unicode-variants", help="Bool. Check unicode consistency (use with --alphabet-format)", action="store_true", default=True)
```
Comment (reuben) [inline util/check_characters.py]: If you make it `action="store_true"` and also `default=True`, it's impossible to set this flag to False.

File: util/check_characters.py [hunk 2]
```diff
@@ -37,7 +40,14 @@ def main():
...
                 for row in reader:
-                    all_text |= set(str(row[2]))
+                    if args.unicode_variants:
+                        unicode_transcript = unicodedata.normalize("NFKC", row[2])
+                        if row[2] != unicode_transcript:
+                            raise ValueError(row[2])
+                    all_text |= set(row[2])
+            except ValueError as e:
+                print("Your input file", in_file, "contains at least one transript with unicode chars on more than one code-point: '{}'".format(e))
+                sys.exit(-1)
```
Comment (reuben) [inline util/check_characters.py]: This error message doesn't tell the user what they should do: NFKC normalization on their data.

File: util/check_characters.py [hunk 3]
```diff
@@ -37,7 +40,14 @@ def main():
             try:
                 next(reader, None)  # skip the file header (i.e. "transcript")
                 for row in reader:
-                    all_text |= set(str(row[2]))
+                    if args.unicode_variants:
+                        unicode_transcript = unicodedata.normalize("NFKC", row[2])
+                        if row[2] != unicode_transcript:
+                            raise ValueError(row[2])
+                    all_text |= set(row[2])
+            except ValueError as e:
```
Comment (reuben) [inline util/check_characters.py]: Why are you raising and catching in the same scope? Just print the message and exit directly in the if on line 46?

File: util/check_characters.py [hunk 4]
```diff
@@ -19,11 +19,14 @@
 import os
 import sys
 
+import unicodedata
```
Comment (reuben) [inline util/check_characters.py]: nit: unnecessary empty line between imports above and this line.

File: util/check_characters.py (hunk 3 above)
Comment (lissyx) [reply util/check_characters.py]: That was just to make the flow nicer

File: util/check_characters.py [hunk 5]
```diff
@@ -18,12 +18,14 @@
...
 import os
 import sys
+import unicodedata
 
 def main():
     parser = argparse.ArgumentParser()
 
     parser.add_argument("-csv", "--csv-files", help="Str. Filenames as a comma separated list", required=True)
     parser.add_argument("-alpha", "--alphabet-format", help="Bool. Print in format for alphabet.txt", action="store_true")
+    parser.add_argument("-unicode", "--unicode-variants", help="Bool. Check unicode consistency (use with --alphabet-format)", type=bool, default=True)
```
Comment (reuben) [inline util/check_characters.py]: nit: This still has the counter-intuitive behavior that to disable it you need to do `-unicode ''`, any string of length > 0 as a value will set it to True, as that's what `bool(string)` does in Python.

**Pull request merged**